Code/Framework/AzCore/AzCore/RTTI/ReflectContext.h

```diff
@@ -225,10 +270,107 @@ namespace AZ
             , m_dataPtr(p) {}
         const T& Get(void* instance) const override { return (reinterpret_cast<C*>(instance)->*m_dataPtr); }
         DataPtr GetMemberDataPtr() const { return m_dataPtr; }
+
+        AZ::Dom::Value GetAsDomValue(void* instance) override
+        {
+            return AZ::Dom::Utils::ValueFromType(Get(instance));
+        }
     private:
         DataPtr m_dataPtr;
     };
 
+    template <typename Arg>
+    bool CanInvokeFromDomArrayEntry(const AZ::Dom::Value& domArray, size_t index)
+    {
+        // Args must be packed as an array
+        if (!domArray.IsArray())
+        {
+            return false;
+        }
+
+        // Unneeded arguments will be discarded, so larger array sizes are OK
+        if (index >= domArray.ArraySize())
+        {
+            return true;
+        }
```

Comment:
Shouldn't this return `false` if the index is out of bounds? I saw the comment, but didn't understand why this would return true

Reply:
A pretty common pattern in the RPE is to have different attribute signatures depending on what you care about, e.g. `PostChangeNotify` might have a signature of either `void(T&)` (with the changed member) or just `void()` (just letting you know something has changed). Rather than making attributes enumerate every possible permutation of these, we instead just allow silently discarding attributes, so we can call `PostChangeNotify(changedThing)` and allow an attribute that implemented in the form of `void()` to work without user intervention.